operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -164,23 +164,38 @@ public static Step createFailureCountStep(V1Job domainIntrospectorJob) {
   static class FailureCountStep extends DomainStatusUpdaterStep {
 
     private final V1Job domainIntrospectorJob;
+    private final boolean resetFailureCount;
```

Comment:
You have changed this step to do one of two things, based on the value of a flag. That is generally poor practice. It would be better simply to add a new step to reset the failure count, if that is needed, and leave the original step alone.

Reply:
ok. I had originally created two steps but later thought that I can reuse the same method and put additional logic in it. I'll change it back to add a new step for resetting failure count and keep the original step same.